kale/loaddata/few_shot.py

```diff
@@ -0,0 +1,107 @@
+import os
+from typing import Any
+
+import numpy as np
+import torch
+from PIL import Image
+from torch.utils.data import Dataset
+
+
+class NWayKShotDataset(Dataset):
+    """
+    It is used to load data for N-way K-shot learning using Prototypical Networks.
+
+    Note:
+        The dataset should be organized as:
+
+        - root
+            - train
+                - class_name 1
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - class_name 2
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - ...
+            - val
+                - class_name 1
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - class_name 2
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - ...
+            - test (optional)
+                - class_name 1
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - class_name 2
+                    - xxx.png
+                    - yyy.png
+                    - ...
+                - ...
+
+    Args:
+        path (string): The root directory of the data.
+        mode (string): The mode of the dataset. It can be 'train', 'val' or 'test'. Default: 'train'.
+        k_shot (int): Number of support examples per class in each episode. Default: 5.
+        query_samples (int): Number of query examples per class in each episode. Default: 15.
```

Comment:
I changed `k_shots`, `k_queries/query_samples`, `n_ways` to `num_support_samples`, `num_query_samples`, `num_classes` so that we can avoid some confusing parts for users and also try to align the names.